### PR TITLE
Feature/sign in/up dialog

### PR DESCRIPTION
**Pull request by piloshachar14**

### CODE REVIEW

File: apps/client/src/components/heading/heading.tsx [hunk 1]
```diff
@@ -1,57 +1,69 @@
...
 import SettingsIcon from '@mui/icons-material/Settings';
+import { useState } from 'react';
+import { SignIn } from '../sign-in-form';
 type Props = {
   isLogin: boolean;
   title: string;
   desc: string;
 };
 export const Heading: React.FC<Props> = ({ title, desc, isLogin }) => {
+  const [dialogOpen, setDialogOpen] = useState(false);
```
Comment (SeanCarmel) [inline apps/client/src/components/heading/heading.tsx]: I think it's better to change dialogOpen to isDialogOpen.

File: apps/client/src/components/heading/heading.tsx [hunk 2]
```diff
@@ -1,57 +1,69 @@
...
+        </div>
+        {!isLogin ? (
+          <LoginIcon
+            sx={{
+              position: 'sticky',
+              color: '#80FFFF',
+              backgroungcolor: BackgroundColor,
+              right: 40,
+              top: 40,
+              fontSize: 70,
```
Comment (SeanCarmel) [inline apps/client/src/components/heading/heading.tsx]: I think when you write without rem/em/%, it considers it as px.

File: apps/client/src/components/leaderboard/leaderboard.module.css [hunk 3]
```diff
@@ -5,3 +5,27 @@
...
+}
+.dividerSilver {
+  height: 100%;
+  width: 0.1em;
+  background-color: var(---silver-leaderboard-color);
+}
+.dividerBronze {
+  height: 100%;
+  width: 0.1em;
+  background-color: var(---bronze-leaderboard-color);
```
Comment (SeanCarmel) [inline apps/client/src/components/leaderboard/leaderboard.module.css:30]: I think it's better to create one class for divider, and a class for each color, and then use them both accordingly.

```suggestion
.divider {
  height: 100%;
  width: 0.1em;
}
.goldBackground{
  background-color: var(---gold-leaderboard-color);
}
{/*...*/}

File: apps/client/src/components/main-page/main-page.module.css [hunk 4]
```diff
@@ -1,15 +1,15 @@
 :root {
-  ---background-image:linear-gradient(147deg, #4d4855 0%, #000000 74%);
-  ---bakcground-color:background-color: #4d4855;
+  ---background-image: linear-gradient(147deg, #4d4855 0%, #000000 74%);
+  ---bakcground-color: #4d4855;
```
Comment (SeanCarmel) [inline apps/client/src/components/main-page/main-page.module.css]: I think ---bakcground-color is unnecessary.

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think it's better to change to: import {...} from '@mui/material'

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+import { ClickAwayListener } from '@mui/base/ClickAwayListener';
+import TextField from '@mui/material/TextField';
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think it's better to change to: import {...} from '@mui/material'

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+import { User } from '../../store';
+import Stack from '@mui/material/Stack';
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think it's better to change to: import {...} from '@mui/material'

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+import { ThemeProvider, Button, createTheme } from '@mui/material';
+
+type Props = {
+  dialogOpen: boolean;
+  setDialogOpen: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const SignIn: React.FC<Props> = ({ dialogOpen, setDialogOpen }) => {
+  const handleClickAway = () => {
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx:21]: I don't understand the meaning of the name `handleClickAway `

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        open={dialogOpen}
+        sx={{
+          direction: 'rtl',
+          width: '100%',
+          height: '100%',
+          position: 'center',
+          alignContent: 'center',
+          '& .MuiPaper-root': {
+            height: '600px',
+            width: '600px',
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think you shouldn't use px

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+          },
+        }}
+      >
+        <DialogTitle
+          sx={{
+            paddingTop: '30px',
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think you shouldn't use px

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+          }}
+          component="h1"
+          align="center"
+        >
+          !ברוכים הבאים
+        </DialogTitle>
+        <ClickAwayListener onClickAway={handleClickAway}>
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I don't know mui ClickAwayListener, I think onClose is better.

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+              color: 'var(--white-color)',
+              '&.MuiDialogContent-root': {
+                height: '80%',
+              },
+            }}
+          >
+            <Stack
+              component="form"
+              sx={{
+                width: '25ch',
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I don't know what is ch, please tell me later.

File: apps/client/src/components/sign-in-form/sign-in.tsx [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+              <Button
+                sx={{
+                  width: '40%',
+                  color: 'var(---white-color)',
+                  transitionDuration: '100ms',
+                  '&:hover': {
+                    cursor: 'pointer',
+                    transform: 'scale(1.2)',
+                  },
+                }}
```
Comment (SeanCarmel) [inline apps/client/src/components/sign-in-form/sign-in.tsx]: I think you should use const for sx beacuse it's repeated.

**Review by SeanCarmel — CHANGES_REQUESTED** [review]
:-)

File: apps/client/src/components/sign-in-form/sign-in.tsx (hunk 13 above)
Comment (piloshachar14) [reply apps/client/src/components/sign-in-form/sign-in.tsx]: nice one!

**Pull request merged**